src/pipeline/batch_inference.py

```diff
@@ -47,6 +47,7 @@ def gen_local_lm_responses(
                 results["target_responses"].append(target_response)
                 results["candidate_responses"].append(lm_response)
 
+    results['candidate_models'] = [model_id] * len(results["instructions"])
```

Comment:
What's the logic behind this?

Also, small formatting nit: not a good practice to mix single and double quotes.

Reply:
This is simply to record what model has generated the outputs.
Thought it might be a good piece of information.

Reply:
Hmm, a better alternative would be to force the users to log the training data to `wandb`, for example and simply appending the run link.

Reply:
Oh this is to record such information to the hf hub.

Since all steps run independently, it would be hard to track down what models output what, which LLM is used to evaluate, etc., 

What do you think?

Reply:
At least the `revision` (SHA) of the model checkpoints?

Reply:
You mean revision info along with the model ID, so two